pymc/gp/hsgp_approx.py

```diff
@@ -372,3 +300,205 @@ def conditional(self, name: str, Xnew: TensorLike, dims: Optional[str] = None):
+    drop_first: bool
+        Default `False`. Sometimes the first basis vector is quite "flat" and very similar to
+        the intercept term.  When there is an intercept in the model, ignoring the first basis
+        vector may improve sampling.
+    scale_func: None, 2D array, or instance of Covariance
+        The covariance function.
+    mean_func: None, instance of Mean
+        The mean function.  Defaults to zero.
+    nu: TensorLike
+        Student T degrees of freedom
+
+    Examples
+    --------
+    .. code:: python
+
+        # A three dimensional column vector of inputs.
+        X = np.random.rand(100, 3)
+
+        with pm.Model() as model:
+            # Specify the covariance (scale) function.
+            # Three input dimensions, but we only want to use the last two.
+            scale_func = pm.gp.cov.ExpQuad(3, ls=0.1, active_dims=[1, 2])
+
+            # Specify the HSTP.
+            # Use 25 basis vectors across each active dimension for a total of 25 * 25 = 625.
+            # The value `c = 4` means the boundary of the approximation
+            # lies at four times the half width of the data.
+            # In this example the data lie between zero and one,
+            # so the boundaries occur at -1.5 and 2.5.  The data, both for
+            # training and prediction should reside well within that boundary.
+            #
+            # Do not forget `nu` this time
+            tp = pm.gp.HSTP(m=[25, 25], c=4.0, scale_func=cov_func, nu=5.)
+
+            # Place a GP prior over the function f.
+            f = tp.prior("f", X=X)
+
+        ...
+
+        # After fitting or sampling, specify the distribution
+        # at new points with .conditional
+        Xnew = np.linspace(-1, 2, 50)[:, None]
+
+        with model:
+            fcond = tp.conditional("fcond", Xnew=Xnew)
+
+    References
+    ----------
+    -   Ruitort-Mayol, G., and Anderson, M., and Solin, A., and Vehtari, A. (2022). Practical
+        Hilbert Space Approximate Bayesian Gaussian Processes for Probabilistic Programming
+
+    -   Solin, A., Sarkka, S. (2019) Hilbert Space Methods for Reduced-Rank Gaussian Process
+        Regression.
+
+    -   Jeremy Sellier, Petros Dellaportas, (2023). Bayesian online change point detection
+        with Hilbert space approximate Student-t process
+    """
+
+    def __init__(
+        self,
+        m: Sequence[int],
+        L: Optional[Sequence[float]] = None,
+        c: Optional[numbers.Real] = None,
+        drop_first: bool = False,
+        parameterization="noncentered",
+        *,
+        mean_func: Mean = Zero(),
+        scale_func: Covariance,
+        nu: pt.TensorLike,
+    ) -> None:
+        super().__init__(
+            m,
+            L,
+            c,
+            drop_first,
+            parameterization,
+            mean_func=mean_func,
+            cov_func=scale_func,
+        )
+        self.nu = pt.as_tensor_variable(nu)
```

Comment:
Referencing variables in the scope of the class is what's messing up with https://github.com/pymc-devs/pymc/issues/6883

Is there another way?

Reply:
> Is there another way?

Yeah, probably for the GP classes it's possible, but I think for the covariance functions is where it will be a lot harder.